src/components/02_atoms/MultipleFields/MultipleFields.js

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import List from '@material-ui/core/List';
+import AddIcon from '@material-ui/icons/Add';
+import Button from '@material-ui/core/Button';
+import ListItem from '@material-ui/core/ListItem';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FormLabel from '@material-ui/core/FormLabel';
+import ReorderIcon from '@material-ui/icons/Reorder';
+import FormControl from '@material-ui/core/FormControl';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+
+const Add = styled('div')`
+  .icon {
+    margin-left: 10px;
+  }
+`;
+
+const style = {
+  listItemStyles: {
+    paddingLeft: 0,
+  },
+  ListItemIconStyles: {
+    cursor: 'move',
+  },
+};
+
+class MultipleFields extends Component {
+  static propTypes = {
+    label: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    component: PropTypes.func.isRequired,
+    value: PropTypes.arrayOf(PropTypes.string).isRequired,
+  };
+
+  /**
+   * Initial state
+   */
+  state = {
+    currentIndex: -1,
+    selectedValue: null,
+  };
+
+  componentDidMount() {
+    const { value, onChange } = this.props;
+    // Set the initial value to an empty string if no values in array.
+    if (!value.length) {
+      onChange(['']);
+    }
+  }
+
+  /**
+   * Sets the state value with the selected element.
+   * @param {Event} event
+   */
+  onDragStart = event => {
+    // setData needed for FireFox, needs to setData to work
+    event.dataTransfer.setData('text', '');
+    event.dataTransfer.effectAllowed = 'move';
+    const currentIndex = parseInt(event.currentTarget.dataset.key, 10);
+
+    this.setState({
+      currentIndex,
+      selectedValue: this.props.value[currentIndex],
+    });
+  };
+
+  /**
+   * Changes the current value with the value that is under the
+   * current value, and replace the over value with the current
+   * value.
+   * @param {Event} event
+   */
+  onDragOver = event => {
+    event.preventDefault();
+    const {
+      props: { value },
+      state: { currentIndex, selectedValue },
+    } = this;
+
+    const overIndex = parseInt(event.currentTarget.dataset.key, 10);
+    if (currentIndex !== overIndex) {
+      // Changes the two values with one another
+      value[currentIndex] = value[overIndex];
+      value[overIndex] = selectedValue;
```

Comment:
Why do we have to store the selectedValue here? I would have expected that instead of selectedValue you could write 
```
selectedValue = value[currentIndex];
value[currentIndex] = value[overIndex];
value[overIndex] = selectedValue;
```

Reply:
A good point, fixed.